ps: clarify grep example (#12650)
tldr-pages/tldr: pages/common/ps.md

`ps auxww`

> - Search for a process that matches a string (Note: This will also match the `grep` process itself when using a simple string.):

Reviewer: ```suggestion
- Search for a process that matches a string (the brackets filter out `grep` itself):
```
If we include the brackets, how about this?

> - Search for a process that matches a string (Note: This will also match the `grep` process itself when using a simple string.):
> 
> `ps aux | grep {{string}}`

Reviewer 2: ```suggestion
- Search for a process that matches a string (enclose the first letter of searched string in `[]` to filter out `grep` itself):

`ps aux | grep {{'[p]ython3'}}`
```

There is actually this clever trick to enclose the first letter in `[]` to filter out `grep` itself. Let's see what other maintainers think about this change.

Here is a demo:

```console
$ ps aux | grep '[s]leep'
user    3610327  0.0  0.0  18180  1920 pts/1    SN+  20:16   0:00 sleep 1000
$
```

Reviewer 3: Or just use `pgrep`?

Reviewer 2: `pgrep` does not show as much info as `ps`. For example, the user who is running a specific process.

Author: > There is actually this clever trick to enclose the first letter in `[]` to filter out `grep` itself. Let's see what other maintainers think about this change.

I had considered this as well, but decided against it because it provides info about `grep` on the page about `ps`. It also doesn't explain _why_ `[t]his` works, and adding an explanation would put even more `grep` info on the `ps` page.

Reviewer: > I had considered this as well, but decided against it because it provides info about `grep` on the page about `ps`. It also doesn't explain _why_ `[t]his` works, and adding an explanation would put even more `grep` info on the `ps` page.

IMO it's fine to use other tools in any way (even a complex one) and don't explain _why_ that works.

After all, users can always consult the grep page for more information, and IMO we shouldn't restrict ourselves on the integration of different commands as long as each page focus on explain its own command.

I personally support adding the brackets.

Author: > [...] we shouldn't restrict ourselves on the integration of different commands as long as each page focus on explain its own command.

Fair point. Though, come to think of it, one could argue that the entire `grep` example is redundant. The first example is `ps aux` and the third example is the exact same thing plus piping to `grep`. It doesn't add any new information about `ps`.

Reviewer: > [...] It doesn't add any new information about `ps`.

Interesting statement. Personallly, I disagree it is redundant, the focus of tldr is _how we can use the program in useful ways_ rather than _how it works_ (man pages already do a good job on explaining each option)

`ps auxww`

> - Search for a process that matches a string (the brackets filter out `grep` itself):

Author: ```suggestion
- Search for a process that matches a string (the brackets will prevent `grep` from matching itself):
```

This phrasing makes it more clear that the brackets are `grep` specific and not `ps` specific.
